src/lang/typing.ml

```diff
@@ -316,10 +314,22 @@ and bind ?(variance = `Invariant) a b =
 
 (** Ensure that the type for the method [l] in [a] is a subtype of the one for the same method in [b]. *)
 and unify_meth a b l =
-  let { meth = l; scheme = s1; json_name = json_name1 } = get_meth l a in
-  let { scheme = s2; json_name = json_name2 } = get_meth l b in
+  let { optional = optional1; meth = l; scheme = s1; json_name = json_name1 } =
+    get_meth l a
+  in
+  let { optional = optional2; scheme = s2; json_name = json_name2 } =
+    get_meth l b
+  in
```

Comment:
This would be more symmetric and safer:
```suggestion
  let { optional = optional1; meth = meth1; scheme = s1; json_name = json_name1 } =
    get_meth l a
  in
  let { optional = optional2; meth = meth2; scheme = s2; json_name = json_name2 } =
    get_meth l b
  in
  assert (meth1 = l && meth2 = l);
```